ringersong/src/main/java/com/RingerSong/free/data/YouTubeMusicRepository.kt

```
@@ -216,4 +217,96 @@ class YouTubeMusicRepository(private val context: Context) {
             0L
         }
     }
+
+    suspend fun downloadTrack(videoId: String): DownloadResult = withContext(Dispatchers.IO) {
```

detekt (1.23.8)

[warning] The function downloadTrack appears to be too complex based on Cognitive Complexity (complexity: 28). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)